### ISSUE TITLE
wrong version of lodash loaded

### ISSUE THREAD

**Issue by OllieNilsen** [wallabyjs/public#319]
I'm running exactly the same codebase as my colleagues, but, for some reason, wallaby seems to be loading the wrong version of lodash on mine. Hence, some tests are failing for me in wallaby, though not in mocha. Any ideas would be appreciated.   I'm pasting in our wallaby.js file below:

<code>
module.exports = () => {
  return {
    files: [
      'api/**/*.*',
      'config/**/*.*',
      'test/helpers/**/*.*',
      'node_modules/removals_dashboard/assets/schema/*'
    ],

    tests: [
      'test/**/*.js',
      '!test/bootstrap.test.js',
      'test/**/*.json',
      '!test/helpers/**.js'
    ],

    bootstrap: wallaby => {
      if (global.sails) return;

      wallaby.delayStart();

      wallaby.testFramework.ui('bdd');
      bootstrap = require('./test/helpers/sharedBootstrap');

      require('magic-globals');
      process.env.NODE_PATH = __base;
      require('module').Module._initPaths();

      var freeport = require('freeport');
      var path = require('path');
      var fs = require('fs');
      var existsSync = fs.existsSync;
      // Replacing buildDictionary.optional to resolve hooks from local node_modules
      var buildDictionary = require('sails/node_modules/sails-build-dictionary');
      var originalOptional = buildDictionary.optional;
      buildDictionary.optional = function () {
        if (~arguments[0].dirname.indexOf('node_modules')) {
          arguments[0].dirname = path.join(wallaby.localProjectDir, 'node_modules');
        }
        return originalOptional.apply(this, arguments);
      };

      // Replacing existsSync to make sails not throw error when it doesn't find node module in wallaby cache.
      // Wallaby will use local node module anyway.
      fs.existsSync = (filePath) => {
        if (filePath && filePath.indexOf(path.join('instrumented', 'node_modules'))) {
          return true;
        }
        return existsSync.apply(this, arguments);
      };
      bootstrap.before(() => wallaby.start());
    },
    env: {
      type: 'node',
      runner: 'node'
    }
  };
};
</code>

**Comment by ArtemGovorov** [comment]
Hmm, interesting issue.
So your colleagues are also running wallaby and it works for them? 
How did you find out that it loads an incorrect `lodash` version?
Could you please try adding 
``` JavaScript
console.log(Object.keys(require.cache).filter(function (p) { 
  return ~p.indexOf('lodash'); 
}));
```
to one of your failing tests and have a look into the Failing Tests tab console output for the test to see where does it load the incorrect version of `lodash` from?

Probably not related to the issue, but could you also move `'test/**/*.json'` from the `tests` list to the `files` list? Could you please also check `'test/**/*.js'` pattern includes nothing but tests (considering the exclusion of the `bootstrap.test.js` and `helpers` folder)? `tests` list should only contain your test files (those with `describe`/`it` in it).

**Comment by josh-egan** [comment]
I just ran into this issue too on a machine at work. I put in the console log statement that you recommended, and found that lodash was installed at 'C:\node_modules\lodash\dist\lodash.js'. I'm not sure how that node_modules folder got there, but there were several packages installed there. I just deleted the 'C:\node_modules' folder, restarted wallaby, and it's working fine now.

This is a shared machine at the office, so I have no idea how a node_modules folder ended up on the root. It is curious though that wallaby would use a 'globally installed' package instead of the locally installed package.
